src/main/java/com/liferay/faces/util/cache/CacheFactory.java

```diff
@@ -0,0 +1,228 @@
+				CacheFactory.class);
+
+		return cacheFactory.<K, V>getCache(initialCapacity, maxCapacity);
+	}
+
+	/**
+	 * Returns a new instance of {@link Cache} from the {@link CacheFactory} found by the {@link
+	 * FactoryExtensionFinder}. The returned instance is designed to be accessed and modified by multiple threads
+	 * concurrently, so it is guaranteed to be {@link java.io.Serializable}.
+	 *
+	 * @param  <K>              The type of the cache's keys.
+	 * @param  <V>              The type of the cache's values.
+	 * @param  externalContext  The external context associated with the current faces context.
+	 */
+	public static <K, V> Cache<K, V> getConcurrentCacheInstance(ExternalContext externalContext) {
+
+		CacheFactory cacheFactory = (CacheFactory) FactoryExtensionFinder.getFactory(externalContext,
+				CacheFactory.class);
+
+		return cacheFactory.<K, V>getConcurrentCache();
+	}
+
+	/**
+	 * Returns a new instance of {@link Cache} from the {@link CacheFactory} found by the {@link
+	 * FactoryExtensionFinder}. The returned instance is designed to be accessed and modified by multiple threads
+	 * concurrently, so it is guaranteed to be {@link java.io.Serializable}.
+	 *
+	 * @param  <K>              The type of the cache's keys.
+	 * @param  <V>              The type of the cache's values.
+	 * @param  externalContext  The external context associated with the current faces context.
+	 * @param  initialCapacity  The initial capacity of the cache.
+	 */
+	public static <K, V> Cache<K, V> getConcurrentCacheInstance(ExternalContext externalContext, int initialCapacity) {
+
+		CacheFactory cacheFactory = (CacheFactory) FactoryExtensionFinder.getFactory(externalContext,
+				CacheFactory.class);
+
+		return cacheFactory.<K, V>getConcurrentCache(initialCapacity);
+	}
+
+	/**
+	 * Returns a new instance of {@link Cache} from the {@link CacheFactory} found by the {@link
+	 * FactoryExtensionFinder}. The returned instance is designed to be accessed and modified by multiple threads
+	 * concurrently, so it is guaranteed to be {@link java.io.Serializable}. The returned cache will avoid exceeding the
+	 * maximum cache capacity. The algorithm used to ensure that the cache size does not exceed the max capacity (for
+	 * example least-recently-used) may cause significant performance differences between the returned cache and a cache
+	 * returned from {@link #getConcurrentCache()} or {@link #getConcurrentCache(int)}.
+	 *
+	 * @param  <K>              The type of the cache's keys.
+	 * @param  <V>              The type of the cache's values.
+	 * @param  externalContext  The external context associated with the current faces context.
+	 * @param  initialCapacity  The initial capacity of the cache.
+	 * @param  maxCapacity      The maximum capacity of the cache.
+	 */
+	public static <K, V> Cache<K, V> getConcurrentCacheInstance(ExternalContext externalContext, int initialCapacity,
```

Comment:
@stiemannkj1 Recommend that this method be named `getLRUCacheInstance` for the same reasons above.